src/main/extension/extension_load.cpp

```diff
@@ -168,7 +168,8 @@ static void ComputeSHA256String(const string &to_hash, string *res) {
 	*res = duckdb_mbedtls::MbedTlsWrapper::ComputeSha256Hash(to_hash);
 }
 
-static void ComputeSHA256FileSegment(FileHandle *handle, const idx_t start, const idx_t end, string *res) {
+static void ComputeSHA256FileSegment(QueryContext context, FileHandle *handle, const idx_t start, const idx_t end,
```

Comment:
Maybe let's not pass it here yet and only change it in the function body - or revert to the other `Read`, if we introduce two.